server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -402,130 +412,664 @@ describe("Routerlicious", () => {
 				});
 
 				describe("#submitSignal", () => {
-					it("Single Signal", async () => {
-						const signalContent = "TestSignal";
-
-						const firstSocket = webSocketServer.createConnection();
+					const createSignalPromise = (
+						client: Client,
+						expectedSignalCount: number = 1,
+					): Promise<ISignalMessage[]> => {
+						return new Promise<ISignalMessage[]>((resolve) => {
+							const signals: ISignalMessage[] = [];
+
+							const signalListener = (signal: ISignalMessage) => {
+								// Ignore disconnected signals
+								if (signal.clientId !== null) {
+									client.signalCount++;
+									signals.push(signal);
+								}
```

Comment:
Are there any cases where clientId is null? Can we make that happen and check for it or should we just assert that it is never null?

Reply:
null client id should nack, added a test case for it.